### PR TITLE
Use DATA* data as input instead of void*

### PR DESCRIPTION
**Pull request by AnHeuermann**
Replace void pointer by DATA pointer. Leftover from 8f15490c2345efda8e78beebe7e580d282d0e038.

### Related Issues

Fixes https://github.com/OpenModelica/OpenModelica/issues/9258.

### Purpose

Don't cast a DATA* to void* and back to DATA* on function call.

### CODE REVIEW

File: OMCompiler/SimulationRuntime/c/simulation_data.h [hunk 1]
```diff
@@ -377,10 +377,10 @@ typedef struct LINEAR_SYSTEM_DATA
   void (*setBElement)(int row, double value, void *data, threadData_t *threadData);
 
   analyticalJacobianColumn_func_ptr analyticalJacobianColumn;
-  int (*initialAnalyticalJacobian)(void*, threadData_t*, ANALYTIC_JACOBIAN*);
+  int (*initialAnalyticalJacobian)(DATA* data, threadData_t*, ANALYTIC_JACOBIAN*);
```
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:380]: For consistency you should also name the other arguments here.

File: OMCompiler/SimulationRuntime/c/simulation_data.h [hunk 2]
```diff
@@ -461,7 +461,7 @@ typedef struct STATE_SET_DATA
    * if analyticalJacobianColumn == NULL no analyticalJacobian is available
    */
   analyticalJacobianColumn_func_ptr analyticalJacobianColumn;
-  int (*initialAnalyticalJacobian)(void*, threadData_t*, ANALYTIC_JACOBIAN*);
+  int (*initialAnalyticalJacobian)(DATA* data, threadData_t*, ANALYTIC_JACOBIAN*);
```
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:464]: For consistency you should also name the other arguments here.

File: OMCompiler/SimulationRuntime/c/simulation_data.h [hunk 3]
```diff
@@ -377,10 +377,10 @@ typedef struct LINEAR_SYSTEM_DATA
   void (*setBElement)(int row, double value, void *data, threadData_t *threadData);
 
   analyticalJacobianColumn_func_ptr analyticalJacobianColumn;
-  int (*initialAnalyticalJacobian)(void*, threadData_t*, ANALYTIC_JACOBIAN*);
+  int (*initialAnalyticalJacobian)(DATA* data, threadData_t*, ANALYTIC_JACOBIAN*);
 
   void (*residualFunc)(RESIDUAL_USERDATA* userData, const double* x, double* res, const int* flag);
-  void (*initializeStaticLSData)(void*, threadData_t *threadData, void*, modelica_boolean);
+  void (*initializeStaticLSData)(DATA* data, threadData_t *threadData, void*, modelica_boolean);
```
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:383]: For consistency you should also name the other arguments here.

**Review by anotheruserofgithub — COMMENTED** [review]
[Sorry I did separate reviews instead of a single one...]

File: OMCompiler/SimulationRuntime/c/simulation_data.h (hunk 1 above)
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:380]: For consistency you should also name the other arguments here.

File: OMCompiler/SimulationRuntime/c/simulation_data.h (hunk 3 above)
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:383]: For consistency you should also name the other arguments here.

File: OMCompiler/SimulationRuntime/c/simulation_data.h (hunk 2 above)
Comment (anotheruserofgithub) [inline OMCompiler/SimulationRuntime/c/simulation_data.h:464]: For consistency you should also name the other arguments here.

**Review by anotheruserofgithub — COMMENTED** [review]
Also I see that you are mixing pointer styling (type* var and type *var), though I don't know if this respects some convention.

**Comment by AnHeuermann** [comment]
> Also I see that you are mixing pointer styling (type* var and type *var), though I don't know if this respects some convention.

Yeah, I'm very inconsistent with these. It's a style question and we don't have coding standard for OpenModelica. So I'm using my favorite one and forget to change it for older code.

**Comment by anotheruserofgithub** [comment]
Maybe you could also use this occasion to correct `initSparsPattern` in `initSparsePattern`?

**Comment by AnHeuermann** [comment]
> Maybe you could also use this occasion to correct initSparsPattern in initSparsePattern?

Sure

**Pull request merged**